tests/test_model.py

```diff
@@ -321,49 +321,58 @@ def test_product_stream_builds():
     rhel_8_2_base_build = SoftwareBuildFactory()
     rhel_7_1_build = SoftwareBuildFactory()
     rhel, rhel_7, rhel_7_1, _, rhel_8_1, rhel_8_2, rhel_8_2_base = create_product_hierarchy()
+    # ProductModel.builds now uses components to lookup builds, so we need to create Components
+    # and call save_product_taxonomy on the builds to ensure we have the connection from streams
+    # to builds via components
     ProductComponentRelation.objects.create(
         type=ProductComponentRelation.Type.COMPOSE,
         # This is a product stream ref
         product_ref=rhel_8_2.name,
-        build_id=rhel_8_2_build.build_id,
+        software_build=rhel_8_2_build,
     )
+    c82 = ComponentFactory(software_build=rhel_8_2_build)
+    ComponentNode.objects.create(obj=c82, purl=c82.purl, parent=None)
```

Comment:
Let's please not use cryptic variable names like this. rhel_8_2_component is much easier to understand.
```suggestion
    rhel_8_2_component = ComponentFactory(software_build=rhel_8_2_build)
    ComponentNode.objects.create(obj=rhel_8_2_component, purl=rhel_8_2_component.purl, parent=None)
```

Reply:
I removed these Component references.